src/test/scala/is/hail/methods/SkatSuite.scala

```diff
@@ -206,108 +171,73 @@ class SkatSuite extends SparkSuite {
       .annotateVariantsExpr("va.genes = [v.start % 2, v.start % 3].toSet") // three overlapping genes
   }
   
-  def hailVsRTest(useBN: Boolean, useDosages: Boolean, useLogistic: Boolean, useLargeN: Boolean,
-    displayValues: Boolean = false, tol: Double = 1e-5) {
-   
-    require(useBN || !useLogistic)
+  def hailVsRTest(useBN: Boolean = false, useDosages: Boolean = false, logistic: Boolean = false,
+    forceLargeN: Boolean = false, displayValues: Boolean = false, tol: Double = 1e-5) {
+    
     require(!(useBN && useDosages))
     
     val (vds, singleKey, weightExpr) = if (useBN) (vdsBN, false, None) else (vdsSkat, true, Some("va.weight"))
     
     val hailKT = vds.skat("va.genes", singleKey = singleKey, "sa.pheno", Array("sa.cov.Cov1", "sa.cov.Cov2"),
-      weightExpr, useLogistic, useDosages, useLargeN)
+      weightExpr, logistic, useDosages, forceLargeN=forceLargeN)
 
     hailKT.typeCheck()
     
     val resultHail = hailKT.rdd.collect()
 
     val resultsR = skatInR(vds, "va.genes", singleKey = singleKey, "sa.pheno", Array("sa.cov.Cov1", "sa.cov.Cov2"),
-      weightExpr, useLogistic, useDosages)
+      weightExpr, logistic, useDosages)
 
     var i = 0
     while (i < resultsR.length) {
-      val qstat = resultHail(i).getAs[Double](1)
-      val pval = resultHail(i).getAs[Double](2)
+      val size = resultHail(i).getAs[Int](1)
+      val qstat = resultHail(i).getAs[Double](2)
+      val pval = resultHail(i).getAs[Double](3)
+      val fault = resultHail(i).getAs[Int](4)
 
       val qstatR = resultsR(i).getAs[Double](1)
       val pvalR = resultsR(i).getAs[Double](2)
-      val fault = resultHail(i).getAs[Int](3)
       
       if (displayValues) {
-        println(f"Davies\' Fault: $fault%d")
-        println(f"HAIL SkatStat: $qstat%2.9f  HAIL pVal: $pval")
-        println(f"   R SkatStat: $qstatR     R pVal: $pvalR")
+        println(f"HAIL qstat: $qstat%2.9f  pval: $pval  fault: $fault  size: $size")
+        println(f"   R qstat: $qstatR%2.9f  pval: $pvalR")
       }
 
       assert(fault == 0)
       assert(D_==(qstat, qstatR, tol))
-      assert(math.abs(pval - pvalR) < 2e-6) // R Davies accuracy is only up to 1e-6
+      assert(math.abs(pval - pvalR) < math.max(tol, 2e-6)) // R Davies accuracy is only up to 1e-6
```

Comment:
Here was my reasoning: Since R uses an accuracy of 1e-6 on Davies, we cannot expect the R and Hail p-values to agree to more than 2e-6 (using the default 1e-6 accuracy in Hail), even if the qstats and evals of the gramian are the same to very high precision. tol in this function is for comparing qstat, and it seemed unreasonable to require closer agreement on p than on qstat since p is computed from qstat, so I took max.

I could instead add a p-value tolerance and note it should be at least 1e-6 (since in principal one could test for even closer agreement when ratcheting up the accuracy of Davies in Hail. Empirically, I find that 1e-6 is too strict for comparing qstats with D_==). I'd then set the p-value tolerance default to 1e5.

Would you prefer that?